tools/GenerateCompilationDatabase.ps1

```diff
@@ -0,0 +1,166 @@
+
+foreach ($line in $lines) {
+    if ($line -match 'Target "ClCompile" in file "([^"]+)" from project "([^"]+)"') {
+        $contextPath = Split-Path $matches[2] -parent
+        Write-Host "Context path set to: $contextPath"
+    } elseif ($line -match 'Cl\.exe (.+)$') {
+        $clLine = $matches[1]
+        $args = $clLine -split ' (?=(?:[^"]*"[^"]*")*[^"]*$)'
+
+        $modifiedArgs = @()
+        foreach ($arg in $args) {
+            if ($arg -match '^(.*\.(cpp|c|h))$' -or $arg -match '^"(.*\.(cpp|c|h))"$') {
+                $filePath = $arg.Trim('"')
+                if (-Not ([System.IO.Path]::IsPathRooted($filePath))) {
+                    $filePath = Join-Path $contextPath $filePath
+                }
+                if ($arg.StartsWith('"') -and $arg.EndsWith('"')) {
+                    $modifiedArgs += '"' + $filePath + '"'
+                } else {
+                    $modifiedArgs += $filePath
+                }
+            } else {
+                $modifiedArgs += $arg
+            }
+        }
+
+        $modifiedLine = "Cl.exe " + ($modifiedArgs -join ' ')
+        Add-Content -Path "temp-filtered2.log" -Value $modifiedLine
+    }
+}
+
+Write-Host "Adjusted file paths and generated: $(Get-Location)\temp-filtered2.log"
+
+$ms2ccPath = Join-Path $PSScriptRoot "ms2cc"
+$ms2ccExe = Join-Path $ms2ccPath "ms2cc.exe"
+
+if (-Not (Test-Path $ms2ccExe)) {
+    Write-Host "Downloading ms2cc..."
+    $ms2ccUrl = "https://github.com/freddiehaddad/ms2cc/releases/download/v$Ms2ccVersion/ms2cc-$Ms2ccVersion.zip"
+    $zipPath = Join-Path $PSScriptRoot "ms2cc-$Ms2ccVersion.zip"
+
+    try {
+        Invoke-WebRequest -Uri $ms2ccUrl -OutFile $zipPath -UseBasicParsing
+        Write-Host "Downloaded ms2cc to: $zipPath"
+
+        if (-Not (Test-Path $ms2ccPath)) {
+            New-Item -ItemType Directory -Path $ms2ccPath -Force | Out-Null
+        }
+
+        Expand-Archive -Path $zipPath -DestinationPath $ms2ccPath -Force
+        Write-Host "Extracted ms2cc to: $ms2ccPath"
+
+        Remove-Item $zipPath -Force
+        Write-Host "Cleaned up zip file"
+
+        if (Test-Path $ms2ccExe) {
+            Write-Host "ms2cc successfully downloaded and extracted"
+        } else {
+            Write-Error "Failed to extract ms2cc.exe"
+            exit 1
+        }
+    }
+    catch {
+        Write-Error "Failed to download or extract ms2cc: $_"
+        exit 1
+    }
+} else {
+    Write-Host "ms2cc already exists at: $ms2ccExe"
+}
+
+$backupPath = $null
+
+# If the -Update flag is provided, we want to update an existing compile_commands.json
+# instead of overwriting it.
+if ($Update) {
+    # Save the old compile_commands.json if it exists in a new file
+    $compileCommandsPath = Join-Path (Split-Path $PSScriptRoot -parent) "compile_commands.json"
+    if (Test-Path $compileCommandsPath) {
+        $backupPath = Join-Path (Split-Path $PSScriptRoot -parent) "compile_commands_old.json"
```

Comment:
Don't write non-source to source directories (`tools` is a source directory)

Write to the project's 'temp' directory. See earlier comments re: `Join-Path $root 'temp'`